gcp-instance-reaper/gcp-instance-reaper.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: v1
+description: Automatically clean up GCP instances that do not have the correct labels
+
+parameters:
+  zone:
+    description: The GCP zone to run in
+    default: us-central1-a
+  terminationDateLabel:
+    description: The name of the label to use for determining the termination date
+    default: termination_date
+  lifetimeLabel:
+    description: The name of the label to use for determining the lifetime
+    default: lifetime
+  dryRun:
+    description: True if this workflow should only print the resources it would delete
+    default: 'true'
+
+steps:
+- name: list-instances
+  image: projectnebula/gcp-list-instances
+  spec:
+    google: &google
+      service_account_info: !Connection { type: gcp, name: my-gcp-account }
+      zone: !Parameter zone
+- name: filter-instances
+  image: projectnebula/core:latest-python
+  spec:
+    terminationDateLabel: !Parameter terminationDateLabel
+    lifetimeLabel: !Parameter lifetimeLabel
+    instances: !Output {from: list-instances, name: instances}
+  inputFile: https://raw.githubusercontent.com/puppetlabs/relay-workflows/master/gcp-instance-reaper/filter-instances.py
+- name: delete-approval
+  description: Wait for approval to delete instances 
+  type: approval
+  dependsOn: filter-instances
+  when: 
```

Comment:
@rick-a-lane-ii does this work now?

Reply:
> @rick-a-lane-ii does this work now?

Approvals have been modified to support when conditions. The core of the implementation is in staging and will be deployed to production soon. Currently, approvals will respect the when condition, but still wait on approval (this is unfortunate but anticipated). As expected, there are quite a few other changes that need to be addressed (approvals are currently a one-off implementation across the board, and that by itself needs addressing).

Remaining:
- Conditions should short-circuit on any known false value; it's not doing this currently (this was a likely situation, and has been officially confirmed now that I have inspected the code).
- Confirmation that the UI will respond accordingly to this situation.
- Confirmation that the one-off approval status handling in the API respects this.
- Confirmation that the status handling (in all of the backend logic differentiates between an approval rejection and an approval being skipped due to the when conditions).